source/security/encryption.txt

```diff
@@ -0,0 +1,319 @@
+instead of the Automatic Encryption Shared Library. ``mongocryptd`` comes
+pre-packaged with enterprise builds of MongoDB Server. For instructions on how to
+install and configure ``mongocryptd``, see the :manual:`Install mongocryptd guide
+</core/csfle/reference/install-library/#steps>` in the MongoDB
+{+server-manual+}.
+
+ffi
+~~~
+
+{+odm+} uses the `ffi gem <https://github.com/ffi/ffi>`__ to call functions from
+``libmongocrypt``. Add the gem to your ``Gemfile`` by running the following
+command in your shell:
+
+.. code-block:: bash
+
+   gem 'ffi'
+
+Create a Customer Master Key
+----------------------------
+
+To encrypt and decrypt data, you must first create a Customer Master Key (CMK).
+A CMK is a key that you use to encrypt your :ref:`Data Encryption Key
+<mongoid-dek>` (DEK). Without access to a CMK, your client application cannot decrypt
+DEKs associated with your encrypted data.
+
+You can create a locally-stored key to use as a local CMK for testing purposes
+by running the following Ruby code:
+
+.. code-block:: ruby
+
+   require 'securerandom'
+
+   SecureRandom.hex(48)
+
+.. warning::
+
+   Using a local CMK in a production environment is insecure. For production
+   environments we recommend using a remote key management service to create and
+   store your CMK.
+
+   To learn more about key management service providers, see the :manual:`KMS Providers
+   guide </core/queryable-encryption/fundamentals/kms-providers>` in the MongoDB 
+   {+server-manual+}.
+
+Configure your Client
+---------------------
+
+CSFLE requires additional configuration for your MongoDB client. To configure a
+client for CSFLE, add the following code to your ``mongoid.yml`` file:
+
+.. code-block:: bash
+
+   development:
+     clients:
+       default:
+         uri: <Your connection URI>
+         options:
+           auto_encryption_options: # This key enables automatic encryption
+             key_vault_namespace: 'encryption.__keyVault' # Database and collection in which to store data keys
+             kms_providers: # Tells the driver where to obtain master keys
+               local: # Specifies that the key is local
+                 key: "<Path to your CMK>"
+             extra_options:
+               crypt_shared_lib_path: '<Path to Shared Library>' # Only required for Ruby versions 2.19 or later
+
+.. note::
+
+   Ensure that you replace the placeholders surrounded by brackets (``<>``) in the preceding code
+   example.
+
+.. _mongoid-dek:
+
+Create a Data Encryption Key
+----------------------------
+
+A Data Encryption Key (DEK) is a key that you use to encrypt the fields in your
+documents. MongoDB stores DEKs, encrypted with your CMK, in the Key Vault
+collection as BSON documents. MongoDB can never decrypt the DEKs, as key
+management is client-side and customer controlled.
+
+To create a DEK in {+odm+}, you can use the
+``db:mongoid:encryption:create_data_key`` rake task, as shown in the following
+example:
```

Comment:
```suggestion
To create a DEK in {+odm+} in a Rails application, you can use the
``db:mongoid:encryption:create_data_key`` rake task, as shown in the following
example:
```

Reply:
The original docs don't mention that this is for a Rails app. Is there something specific that makes this for Rails?